### DISCUSSION TITLE
Shouldn't activating copy mode automatically push it to current key table stack?

### DISCUSSION THREAD

**Discussion by schardev** [wezterm/wezterm#2986] [Q&A] [answered]
I was trying to show the current active mode to the `right-status` which can be done by using `window:active_key_table()` but when I activate copy mode the method returns `nil`. Shouldn't it be more fitting to push (and thereby activate) `copy_mode` key table to current stack by default upon activation?

P.S: I'm trying to somewhat mimic [this](https://github.com/twilsoft/wezmode).

**Comment by wez** [comment, marked as answer]
There's a pane-specific segment to the key table stack that wasn't factored into what that method was looking at.  I've fixed that in 59b609f5920e6ae5c6200b2f1425e2565966bb10

It typically takes about an hour before commits are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out. You can find the nightly downloads for your system in the [wezterm installation docs](https://wezfurlong.org/wezterm/installation.html).

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them. 

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Reply by schardev** [reply]
Thanks for the _blazingly fast_ response. Just tested with the fresh Ubuntu nightly image and it's working as expected 👍
